File: Programming Contest/DSA/Chapter 2/Exercise 2-4/OJ/LightOJ/1182.c
#include <stdio.h>

int main()
{
    int t, n, count, i;

    scanf("%d", &t);

    for (i = 1; i <= t; i++) {
        scanf("%d", &n);

        count = 0;

        while (n) {
            n = n & (n - 1);
            count++;
        }

        printf("Case %d: %s\n", i, (count % 2 == 1) ? "odd" : "even");
    }

    return 0;
}

File: Programming Contest/DSA/Chapter 2/Exercise 2-4/OJ/LightOJ/1182.py
for i in range(1, int(input()) + 1):
    n = int(input())
    b = list(bin(n))
    
    cnt = b.count('1')
    
    if (cnt % 2 == 0):
        print(f"Case {i}: even")
    else:
        print(f"Case {i}: odd")